File: app/src/main/java/gregrdm/githubuserlistkotlin/dagger/NetComponent.java
package gregrdm.githubuserlistkotlin.dagger;

/**
 *
 */

import javax.inject.Singleton;

import dagger.Component;
import gregrdm.githubuserlistkotlin.utils.SchedulersImpl;
import retrofit2.Retrofit;

@Singleton
@Component(modules = {AppModule.class, NetModule.class})
public interface NetComponent {
    Retrofit provideRetrofit();
    SchedulersImpl provideSchedulers();
}

File: app/src/main/java/gregrdm/githubuserlistkotlin/dagger/NetComponent.kt
package gregrdm.githubuserlistkotlin.dagger

import javax.inject.Singleton

import dagger.Component
import gregrdm.githubuserlistkotlin.ui.MainActivity

/**
 *
 */

@Singleton
@Component(modules = arrayOf(AppModule::class, NetModule::class))
interface NetComponent {
    fun inject(activity: MainActivity)
}
